src/checkmatelib/client.py

```diff
@@ -68,10 +75,30 @@ def check_url(  # pylint: disable=inconsistent-return-statements
         response.raise_for_status()
 
         if response.status_code == 204:
+            # No news is good news
             return None
 
         try:
             return BlockResponse(response.json())
 
         except ValueError as err:
             raise CheckmateServiceError("Unprocessable JSON response") from err
+
+    @classmethod
+    def _clean_url(cls, url):
+        """Clean the URL before we send it.
+
+        Checkmate will do this for itself, and doesn't trust our input, but
+        this allows us to fail fast for bad URLs, before Checkmate has a chance
+        to check them. We can also apply extra constraints.
+        """
+
+        # Truncate extremely long URLs, so we don't get 400's and fail open
```

Comment:
Where would the 400 come from if the URL was not truncated here?

Reply:
One of NGINX / gunicorn / Cloudflare will eventually complain if you have an absurdly long URL